### PR TITLE
Add docs for new SqlTypes APIs

### PR DESCRIPTION
**Pull request by David-Engel**
## Summary

Related to:
https://github.com/dotnet/runtime/pull/72724
https://github.com/dotnet/runtime/issues/51836

Partially addresses: https://github.com/dotnet/runtime/issues/76049

### CODE REVIEW

**Comment by gewarren** [comment]
Close/reopen for build.

File: xml/System.Data.SqlTypes/SqlBinary.xml [hunk 1]
```diff
@@ -1920,10 +1920,17 @@
         <Parameter Name="bytes" Type="System.Byte[]" Index="0" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="bytes">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="bytes">The array of bytes to be used to create the <see cref="T:System.Data.SqlTypes.SqlBinary" /> structure.</param>
+        <summary>Turns an array of bytes into a <see cref="T:System.Data.SqlTypes.SqlBinary" /> structure.</summary>
```
Comment (gewarren) [inline xml/System.Data.SqlTypes/SqlBinary.xml:1924]: Converts?

File: xml/System.Data.SqlTypes/SqlDecimal.xml [hunk 2]
```diff
@@ -4110,9 +4110,9 @@ This member is an explicit interface member implementation. It can be used only
         <Parameter Name="destination" Type="System.Span&lt;System.UInt32&gt;" Index="0" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="destination">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="destination">A <see cref="T:System.Span`1" /> of 32-bit unsigned integers.</param>
+        <summary>Writes the four 32-bit unsigned integer values that make up the <see cref="T:System.Data.SqlTypes.SqlDecimal" /> structure into the <see cref="T:System.Span`1" /> parameter provided.</summary>
```
Comment (gewarren) [inline xml/System.Data.SqlTypes/SqlDecimal.xml:4114]:
```suggestion
        <summary>Writes the four 32-bit unsigned integer values that make up the <see cref="T:System.Data.SqlTypes.SqlDecimal" /> structure into the <see paramref="destination" /> span.</summary>
```

File: xml/System.Data.SqlTypes/SqlMoney.xml [hunk 3]
```diff
@@ -769,8 +776,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Returns the Tabular Data Stream (TDS) long value that makes up the structure of the <see cref="T:System.Data.SqlTypes.SqlMoney" />.</summary>
```
Comment (gewarren) [inline xml/System.Data.SqlTypes/SqlMoney.xml:779]:
```suggestion
        <summary>Returns the Tabular Data Stream (TDS) long value that makes up the structure of the <see cref="T:System.Data.SqlTypes.SqlMoney" /> instance.</summary>
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Data.SqlTypes/SqlBinary.xml (hunk 1 above)
Comment (David-Engel) [inline xml/System.Data.SqlTypes/SqlBinary.xml:1924]:
```suggestion
        <summary>Converts an array of bytes into a <see cref="T:System.Data.SqlTypes.SqlBinary" /> structure.</summary>
```

File: xml/System.Data.SqlTypes/SqlDecimal.xml [hunk 4]
```diff
@@ -4110,9 +4110,9 @@ This member is an explicit interface member implementation. It can be used only
         <Parameter Name="destination" Type="System.Span&lt;System.UInt32&gt;" Index="0" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="destination">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="destination">A <see cref="T:System.Span`1" /> of 32-bit unsigned integers.</param>
+        <summary>Writes the four 32-bit unsigned integer values that make up the <see cref="T:System.Data.SqlTypes.SqlDecimal" /> structure into the <see paramref="destination" /> span.</summary>
```
Comment (gewarren) [inline xml/System.Data.SqlTypes/SqlDecimal.xml:4114]:
```suggestion
        <summary>Writes the four 32-bit unsigned integer values that make up the <see cref="T:System.Data.SqlTypes.SqlDecimal" /> structure into the <paramref name="destination" /> span.</summary>
```

**Pull request merged**